### PR TITLE
[TRANSACTIONS] Endpoint Update Transaction

### PR DESCRIPTION
**Pull request by Jair-Academlo**
AS: admin
I WANT TO  update  an transaction by id
SO I can update information the transaction by id

Acceptance Criteria:

PUT /transactions/:id
- Should valid if the is existing transaction in DB by sent id
- The request should be contain user, category, amount, and date fields

## Evidence:

update dates success!

Actualizar_Transaccion

in case error

Screenshot_20221104_175942

### CODE REVIEW

File: controllers/Transactions/transationsUpdateController.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+module.exports = {
+  updateTransactionById: catchAsync(async (req, res, next) => {
+    try {
+      const { id } = req.params;
+
+      const transactionById = await Transaction.findOne({ where: { id } });
+
+      if (!transactionById) {
+        throw next(new ErrorObject(" transaction by Id not found ", 404));
```
Comment (miguelbel00) [inline controllers/Transactions/transationsUpdateController.js:15]: throw the error outside the next function , next function is used in catch

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "url": "https://github.com/gabodmc/nodejs-boilerplate"
   },
   "scripts": {
-    "start": "node app.js",
+    "start": "nodemon app.js",
```
Comment (miguelbel00) [inline package.json:13]: nodemon its not necessary

**Review by miguelbel00 — CHANGES_REQUESTED** [review]

File: controllers/Transactions/transationsUpdateController.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+      endpointResponse({
+        res,
+        message: "result successfully",
+        body: transactionById,
+      });
+    } catch (error) {
+      const httpError = createHttpError(
+        error.statusCode,
+        `[Error retrieving index] - [index - GET]: ${error.message}`
```
Comment (miguelbel00) [inline controllers/Transactions/transationsUpdateController.js:28]: change the example error for a custom error

File: controllers/Transactions/transationsUpdateController.js (hunk 1 above)
Comment (Jair-Academlo) [reply controllers/Transactions/transationsUpdateController.js]: ya se resolvieron los cambios

File: package.json (hunk 2 above)
Comment (Jair-Academlo) [reply package.json]: ya se resolvieron los cambios

File: controllers/Transactions/transationsUpdateController.js (hunk 3 above)
Comment (Jair-Academlo) [reply controllers/Transactions/transationsUpdateController.js]: ya se resolvieron los cambios

File: controllers/Transactions/transationsUpdateController.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+      endpointResponse({
+        res,
+        message: "result successfully",
+        body: transactionById,
+      });
+    } catch (error) {
+      const httpError = createHttpError(
+        error.statusCode,
+        `[Error retrieving Transactions] - [index - PUT]: ${error.message}`
```
Comment (miguelbel00) [inline controllers/Transactions/transationsUpdateController.js:28]: change index for transactionsController

**Review by miguelbel00 — CHANGES_REQUESTED** [review]

File: controllers/Transactions/transationsUpdateController.js (hunk 4 above)
Comment (Jair-Academlo) [reply controllers/Transactions/transationsUpdateController.js]: ya quedo realizado el cambio

**Review by miguelbel00 — APPROVED** [review]
All Ok

**Review by NandierSR26 — COMMENTED** [review]
En la ruta PUT, se encuentra un doble slash
image
deberia ir solo uno, o da  igual?

**Pull request merged**